config/src/main/java/org/springframework/security/config/annotation/web/configurers/SessionManagementConfigurer.java

```diff
@@ -570,7 +571,9 @@ private SessionAuthenticationStrategy getSessionAuthenticationStrategy(H http) {
 			SessionRegistry sessionRegistry = getSessionRegistry(http);
 			ConcurrentSessionControlAuthenticationStrategy concurrentSessionControlStrategy = new ConcurrentSessionControlAuthenticationStrategy(
 					sessionRegistry);
-			concurrentSessionControlStrategy.setMaximumSessions(this.maximumSessions);
+			if (this.sessionLimitStrategy != null) {
```

Comment:
Good point. I think the issue here is that the code is trying to use the field for two different things; determine if concurrency control is enabled and determine the maximum number of sessions.

It may work better to separate the two like so:

```java
private SessionLimitStrategy maximumSessions = SessionLimitStrategy.of(1)

private boolean concurrentControlEnabled;
```